stock_picking_batch_actual_date/models/stock_picking_batch.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockPickingBatch(models.Model):
+    _name = "stock.picking.batch"
+    _inherit = ["stock.picking.batch", "stock.actual.date.mixin"]
+
+    def _get_stock_moves(self):
+        # Return an empty recordset to prevent duplicate `actual_date`
+        # assignments on pickings' stock moves
+        return self.env["stock.move"]
+
+    def action_done(self):
+        self.ensure_one()
+        self.picking_ids.write({"actual_date": self.actual_date})
+        return super().action_done()
```

Comment:
Some pickings can be detached in the process of `action_done()`. Please add a test case for this.
```suggestion
        picks_before_done = self.picking_ids
        res = super().action_done()
        picks_detached = picks_before_done - self.picking_ids
        picks_detached.write({"actual_date": False})
        self.picking_ids.filtered(lambda x: x.actual_date != self.actual_date).write({"actual_date": self.actual_date})
        return res
```